core/graphelier-service/utils/utils.go

```diff
@@ -1,9 +1,31 @@
 package utils
 
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
 // Abs : Default Abs func for golang takes a float and returns a float. Need one for ints
 func Abs(n int64) int64 {
 	if n < 0 {
 		return -n
 	}
 	return n
 }
+
+func DebugTimer(call interface{}) func() {
+	start := time.Now()
+	return func() {
+		log.Debugf("%s took %v\n", call, time.Since(start))
+	}
+}
+
+func TraceTimer(call interface{}) func() {
```

Comment:
How come you made `call` a generic interface rather than a string?

Reply:
The idea was to make it more generic where a struct could be passed for printing, though I realize the format string uses `%s` and therefore can only accept strings. I think I will change the `interface` parameter to string and offload the formatting work to the caller of Timer.